vllm/compilation/fusion.py

```diff
@@ -10,6 +10,29 @@
 
 logger = init_logger(__name__)
 
+@torch.library.custom_op("neuralmagic::silu_mul_quant", mutates_args=())
+def silu_mul_quant(x: torch.Tensor, scale: torch.Tensor) -> torch.Tensor:
+    result = torch.empty(x.shape[0], x.shape[1] // 2, device=x.device, dtype=torch.float8_e4m3fn)
+
+    torch.ops._C.silu_and_mul_quant(result, x, scale)
+    return result
```

Comment:
The double dispatching might be hurting us here.  Is there some reason we can't just inline this pattern into the rpelacement?

Reply:
I'll let @sage confirm but I think this is hiding the slice issue

Reply:
Or see if we can call `torch.ops._C.silu_and_mul_quant.default` directly?  (if we make sure that this only gets applied for CUDA)